moesifwsgi/send_async_event.py

```diff
@@ -0,0 +1,44 @@
+
+class SendEventAsync:
+    def exit_handler(self, scheduler, debug):
+        try:
+            # Shut down the scheduler
+            scheduler.shutdown()
+        except:
+            if debug:
+                print("Error while closing the queue or scheduler shut down")
+
+    def send_event(self, api_client, batch_events, debug):
+        try:
+            if debug:
+                print("Sending events to Moesif")
+            batch_events_api_response = api_client.create_events_batch(batch_events)
+            if debug:
+                print("Events sent successfully")
+            # Fetch Config ETag from response header
+            batch_events_response_config_etag = batch_events_api_response.get("X-Moesif-Config-ETag")
+            # Return Config Etag
+            return batch_events_response_config_etag
+        except Exception as ex:
+            if debug:
+                print("Error sending event to Moesif")
+                print(str(ex))
+            return None
+
+    def async_client_create_event(self, api_client, moesif_events_queue, debug, batch_size):
+        batch_events = []
+        try:
+            while moesif_events_queue.qsize() > 0:
```

Comment:
similar to comment i had in another pr..this can be optimised, first read events until either moesif_events_queur is empty or batchsize, then outside of while loop call once to self.send_events if batch_events is not empty

Reply:
also i dont know how it is async..its more like send whatever events available now

Reply:
Have optimized the function and rename the file and function name. It's batching and sending events.